### PR TITLE
Consider the 'use-default' option if present, to avoid using a GtkDialog

### PR DESCRIPTION
**Pull request by mariospr**
This addresses issue https://github.com/flatpak/xdg-desktop-portal/issues/10

### CODE REVIEW

**Comment by matthiasclasen** [comment]
As commented on the other branch: I think the frontend should just skip the appchooser in this case.

What we do want in the backend, though, is preselecting the last choice.

**Comment by mariospr** [comment]
Thanks for the feedback. As mentioned in https://github.com/flatpak/xdg-desktop-portal/pull/19#issuecomment-228174576, I have a doubt on how exactly you'd skip the appchooser in this case... would you make call a interface different than org.freedesktop.impl.portal.AppChooser (e.g. org.freedesktop.impl.portal.OpenURI)?

If that's the case then I'm having trouble to see how I can do it without contradicting the suggestion of moving most of the logic to the backend, so I'm probably getting something wrong :)

**Comment by matthiasclasen** [comment]
Did my comment over there clarify things ?

**Comment by mariospr** [comment]
It did, thanks!

**Comment by mariospr** [comment]
PR updated according to the changes done in https://github.com/flatpak/xdg-desktop-portal/pull/19

File: appchooserdialog.c [hunk 1]
```diff
@@ -240,6 +241,10 @@ app_chooser_dialog_new (const char **choices,
       row = GTK_WIDGET (app_chooser_row_new (info));
       gtk_widget_set_visible (row, i < 3 || n_choices < 5);
       gtk_list_box_insert (GTK_LIST_BOX (dialog->list), row, -1);
+
+      /* Activate the default choice if we have one */
+      if (g_strcmp0 (choices[i], default_id) == 0)
+        g_signal_emit_by_name (GTK_LIST_BOX (dialog->list), "row-activated", row, dialog);
```
Comment (matthiasclasen) [inline appchooserdialog.c]: Hmm, I think I'd prefer to add an app_chooser_dialog_set_selected()

**Comment by mariospr** [comment]
Thanks, just updated the PR to put that logic inside a _set_selected() method.

**Pull request merged**